Auto

ConradSkelly opened the pull request:
# Pull request for autos choreo backend

## Included Features:

auto choero backend

## Checklist:

- [x] The code compiles and runs without errors
- [x] The code has been self or peer-reveiwed
- [x] The code has been tested & works as intended

ConradSkelly committed:
made branch

ConradSkelly committed:
Merge remote-tracking branch 'remotes/origin/dev' into auto

ConradSkelly committed:
made auto backend with choreo

ConradSkelly committed:
made choreo backend

ConradSkelly committed:
stuff

eskiles commented on src/main/java/frc/robot/subsystems/Swerve.java:
```diff
@@ -89,22 +96,40 @@ public Swerve() {
+            sample.omega
+                + headingController.calculate(pose.getRotation().getRadians(), sample.heading));
+
+    driveRobotRelative(speeds);
```
This is supposed to call "driveFieldRelative(speeds)", not "driveRobotRelative(speeds)".
The sample is an X, Y, Heading on the field from the trajectory. From the pose, it is getting the field relative X and Y positions to calculate the field relative chassis speeds.

ConradSkelly committed:
found out error with sim drive conflicting with non sim piriodic

ConradSkelly committed:
made it move and run in auto not setting intial position properly

eskiles committed:
Updates to get teleop working with Red/Blue.
Added resetOdometry command to the Auto Command.
Ran the "isSimulated" flag through to the swerve modules. For some reason the updates in periodic are causing the simulated robot to do nothing.

eskiles committed:
Accounting for alliance color should only be in commands from the driver, not trajectories.
